.github/workflows/backup-kubernetes-databases-inhouse.yml

```
@@ -90,14 +106,54 @@ jobs:
 
       - name: Run database backup
         id: dbbackup
-        run: |    
+        run: |
+          if [[ ${{ inputs.part }} -le 0 || ${{ inputs.ratio }} -le 0 ]]; then
+            echo "Error: Both part and ratio must be greater than 0";
+            exit 1;
+          fi
+          
+          if [[ ${{ inputs.part }} -gt ${{ inputs.ratio }} ]]; then
+             echo "Error: part (${{ inputs.part }}) is bigger then ratio (${{ inputs.ratio }})";
+             exit 1;
+          fi
+          
           echo "Starting..."
           backup="${{ inputs.backup }}"
           echo "We are going to backup: ${backup}"
           
           # we have different selects for postgress apps and for wordpress
           if [[ "$backup" == "shared-wordpress" ]]; then label="app.kubernetes.io/name=wordpress"; else label="service=database"; fi
-          select=$(kubectl get pod --namespace "${{ inputs.namespace }}" --insecure-skip-tls-verify --request-timeout=5s --selector "$label" --output jsonpath='{.items..metadata.name}')
+          preselect=$(kubectl get pod --namespace "${{ inputs.namespace }}" --insecure-skip-tls-verify --request-timeout=5s --selector "$label" ${{ inputs.field-selector }} --output jsonpath='{.items..metadata.name}')
+          
+          # Option to process just part of the pods by providing two variables
+          # ratio - split to results to X parts
+          # part - which part should be displayed
+          #
+          # Like to process first half of results, you set ratio to 2 and part to 1. To proceess second half of that set part to 2
+          # If you dont want to split the results, set ratio to 1 and part to 1
+          
+          # count number of pods
+          count=$(echo $preselect | wc -w)
+          echo "Number of all pods: ${count}"
+             
+          # which part from splits to display     
+          part=${{ inputs.part }}    
+          echo "Part: ${part}}"     
+               
+          # ratio for split     
+          ratio=${{ inputs.ratio }}
+          echo "Ratio: ${ratio}"
+          
+          # calculate chunk size
+          chunk=$(( (count + ratio - 1) / ratio ))
+          echo "Chunk: ${chunk}"
+         
+          # Calculate how many pods to skip based on chunk size
+          skip=$(( (part - 1) * chunk ))
+          echo "Skip: ${skip}"
+          
+          # select pods based on chunksize and skip     
+          select=$(echo $preselect | tr ' ' '\n' | tail -n +$((skip + 1)) | head -n $chunk)
```

You don't have to pipe to `tr  ' ' '\n'`. The reason why you don't have new lines is missing quotes around variable. You should be able to just do 
```suggestion
          select=$(echo "$preselect" | tail -n +$((skip + 1)) | head -n $chunk)
```

---

oki thanks, will update

---

i was testing it now, but it didnt worked, even with the quotes on $preselect, list of pods was without a new lines, so tail and head didnt worked. I added the tr back to pipe